frame/contracts/src/lib.rs

```diff
@@ -136,6 +136,43 @@ type BalanceOf<T> =
 /// The current storage version.
 const STORAGE_VERSION: StorageVersion = StorageVersion::new(6);
 
+/// The address generator function used to determine the address of a contract.
+pub trait AddressGenerator<T: frame_system::Config> {
+	fn generate_address(
```

Comment:
Every public item needs rust doc.

Reply:
Fixed

Reply:
Thanks. I think we need to document some warnings here though:
1. Implementer must make sure that there are no collisions (same output for different inputs).
2. Implementer must make sure that the same inputs lead to the same output (including environmental inputs like your storage location).
3. Changing the implementation through a runtime upgrade without a proper storage migration would be a a catastrophic event.

Reply:
Thanks for the suggestion. Appended them to the doc.